### PR TITLE
Move ParseProtobufFromLargeString to proto_utils

### PR DESCRIPTION
**Pull request by Yangqing**
WIP - kicking off contbuild.

### CODE REVIEW

File: caffe2/utils/proto_utils.cc [hunk 1]
```diff
@@ -156,6 +166,14 @@ bool ParseFromString(const string& spec, Message* proto) {
 }
 } // namespace TextFormat
 
+bool ParseProtobufFromLargeString(const string& str, Message* proto) {
+  ::google::protobuf::io::ArrayInputStream input_stream(str.data(), str.size());
```
Comment (orionr) [inline caffe2/utils/proto_utils.cc]: Nit: do you want to add and use

```
using ::google::protobuf::io::ArrayInputStream;
using ::google::protobuf::io::CodedInputStream;
```
above and get rid of the long references here?

File: caffe2/utils/proto_utils.cc [hunk 2]
```diff
@@ -156,6 +166,14 @@ bool ParseFromString(const string& spec, Message* proto) {
 }
 } // namespace TextFormat
 
+bool ParseProtobufFromLargeString(const string& str, Message* proto) {
```
Comment (orionr) [inline caffe2/utils/proto_utils.cc]: Nit: Any chance you want to call this `ParseProtoFromLargeString` to match naming in here?

**Review by orionr — APPROVED** [review]
Minor suggestions, but otherwise looks good!

File: caffe2/utils/proto_utils.cc (hunk 1 above)
Comment (Yangqing) [reply caffe2/utils/proto_utils.cc]: It's only used once so I'll probably just keep it this way?

File: caffe2/utils/proto_utils.cc (hunk 2 above)
Comment (Yangqing) [reply caffe2/utils/proto_utils.cc]: Good point, will do.

**Comment by orionr** [comment]
@caffe2bot retest this please

**Comment by Yangqing** [comment]
Making informed guess that the test failures are orthogonal and addressed in #2359 . Merging this...

**Comment by orionr** [comment]
@caffe2bot retest this please

**Pull request merged**